### PR TITLE
Resolve US26-29 issues

### PR DESCRIPTION
**Pull request by austincarrjones**
## Describe Changes Made
US26-29 were completed async between Austin (26) and Jan (27-29). Jan never made a PR, but he and I pulled his branch down (or main in his case) to try and resolve the merge stuff locally. The main problem was functionality in the Update action on the Admin/Merchants controller. Merging was really problematic. Had to break it all down with Arty. Updated some tests to be more robust as well. Still some minor refactoring possible, left a few comments in there. I didn't commit much while working, but can explain everything.

## If PR is linked to a Issue, Add Issue
Going to update the project board manually this time.

## PR Checklist
- [X] Added Reviewer
- [X] Followed TDD, And Test are Passing
- [ ] Linked Issues for Project Board

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class Admin::MerchantsController < ApplicationController
   def index
-    @merchants = Merchant.all
+    @merchants = Merchant.all 
+    #Best practice is to do 1 variable here. The below variables are unnecessary - just call the model methods(.enabled_merchants) on @merchants in the view
+    #let's refactor in another PR tomorrow.
```
Comment (JCL461437) [inline app/controllers/admin/merchants_controller.rb:5]: Oh! Sounds good!

File: app/controllers/admin/merchants_controller.rb [hunk 2]
```diff
@@ -11,15 +15,33 @@ def edit
...
+      # Not Working, needs to be refactored
+      flash.now[:notice] = 'Please fill out all the required fields.'
+      render :new
+    end
   end
 
-  private
-  def merchant_params
-    params.require(:merchant).permit(:name)
+  def update
```
Comment (JCL461437) [inline app/controllers/admin/merchants_controller.rb:35]: Austin I can't believe you figured this out. Your a legend. It is interesting how much leaner the method is than what we created earlier.

File: spec/models/merchant_spec.rb [hunk 3]
```diff
@@ -8,110 +8,139 @@
...
-      #     create(:transaction, invoice: invoice_8, result: 1)
-      #     create(:transaction, invoice: invoice_9, result: 1)
-
-      #     expect(merchant.count_successful_transactions(customer_1)).to eq(2)
-      #     expect(merchant.count_successful_transactions(customer_2)).to eq(1)
-      #     expect(merchant.count_successful_transactions(customer_6)).to eq(0)
-      #   end
-      # end
+    end
+    # describe "#count_successful_transactions" do
```
Comment (JCL461437) [inline spec/models/merchant_spec.rb:81]: These look great

**Review by JCL461437 — APPROVED** [review]
Great work Austin. Really appreciate all the care and effort

**Review by LancePants97 — APPROVED** [review]

**Review by GBowman1 — APPROVED** [review]

**Pull request merged**